vault/data_source_azure_access_credentials_test.go

```diff
@@ -0,0 +1,74 @@
+package vault
+
+import (
+	"strings"
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-sdk/helper/acctest"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/resource"
+)
+
+func TestAccDataSourceAzureAccessCredentials_basic(t *testing.T) {
+	// This test takes a while because it's testing a loop that
+	// retries real credentials until they're eventually consistent.
+	if testing.Short() {
+		t.SkipNow()
+	}
+	mountPath := acctest.RandomWithPrefix("tf-test-azure")
+	subscriptionID, tenantID, clientID, clientSecret, scope := getTestAzureCreds(t)
+	resource.Test(t, resource.TestCase{
+		Providers: testProviders,
+		PreCheck:  func() { testAccPreCheck(t) },
+		Steps: []resource.TestStep{
```

Comment:
A failure case would be good too. Maybe set up an impossible to satisfy set of requirements (1000 successes in 5 seconds or something like that) and make sure it fails.

Reply:
Just added this: https://github.com/terraform-providers/terraform-provider-vault/pull/713/commits/529625c5f98f49425be1782bcbd4716ff23eaf56

Test output remains as:
```
=== RUN   TestAccDataSourceAzureAccessCredentials_basic
--- PASS: TestAccDataSourceAzureAccessCredentials_basic (25.37s)
PASS
```